src/AdoNet/Orleans.Persistence.AdoNet/Options/AdoNetGrainStorageOptions.cs

```diff
@@ -0,0 +1,114 @@
+
+        /// <summary>
+        /// Stage of silo lifecycle where storage should be initialized.  Storage must be initialzed prior to use.
+        /// </summary>
+        public int InitStage { get; set; } = DEFAULT_INIT_STAGE;
+        /// <summary>
+        /// Default init stage in silo lifecycle
+        /// </summary>
+        public const int DEFAULT_INIT_STAGE = SiloLifecycleStage.ApplicationServices;
+
+        /// <summary>
+        /// The default ADO.NET invariant used for storage if none is given. 
+        /// </summary>
+        public const string DEFAULT_ADONET_INVARIANTS = AdoNetInvariants.InvariantNameSqlServer;
+        /// <summary>
+        /// The invariant name for storage
+        /// </summary>
+        public string AdoInvariant { get; set; } = DEFAULT_ADONET_INVARIANTS;
+
+        #region json serialization related settings
+        /// <summary>
+        /// Whether storage string payload should be formatted in JSON.
+        /// </summary>
+        public bool UseJsonFormat { get; set; }
+        public bool UseFullAssemblyNames { get; set; }
+        public bool IndentJson { get; set; }
+        public TypeNameHandling? TypeNameHandling { get; set; }
+        #endregion
+        /// <summary>
+        /// Whether storage string payload should be formatted in Xml.
+        /// </summary>
+        public bool UseXmlFormat { get; set; }
+    }
+    public class AdoNetGrainStorageOptionsValidator : IConfigurationValidator
+    {
+        private readonly AdoNetGrainStorageOptions options;
+        private readonly string name;
+        public AdoNetGrainStorageOptionsValidator(AdoNetGrainStorageOptions configurationOptions, string name)
+        {
+            this.options = configurationOptions;
+            this.name = name;
+        }
+
+        public void ValidateConfiguration()
+        {
+            if (string.IsNullOrWhiteSpace(this.options.ConnectionString))
+            {
+                throw new OrleansConfigurationException($"Invalid AdoNetGrainStorageOptions for AdoNetGrainStorage {name}. ConnectionString is required.");
+            }
+            if (options.UseXmlFormat&&options.UseJsonFormat)
+            {
+                throw new OrleansConfigurationException($"Invalid AdoNetGrainStorageOptions for AdoNetGrainStorage {name}. Only one serializer and deserializer should be given.");
+            }
+        }
+    }
+
+    public class AdoNetStorageOptionsFormatterResolver : IOptionFormatterResolver<AdoNetGrainStorageOptions>
+    {
+        private IOptionsSnapshot<AdoNetGrainStorageOptions> optionsSnapshot;
+        public AdoNetStorageOptionsFormatterResolver(IOptionsSnapshot<AdoNetGrainStorageOptions> optionsSnapshot)
+        {
+            this.optionsSnapshot = optionsSnapshot;
+        }
+
+        public IOptionFormatter<AdoNetGrainStorageOptions> Resolve(string name)
+        {
+            return new AzureTableStorageOptionsFormatter(name, optionsSnapshot.Get(name));
```

Comment:
What does `AzureTableStorageOptionsFormatter` do here?

Reply:
it format the option settings to an array and would be later logged out when silo start up.